autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/service/AutofillSimpleCredentialsListFragment.kt

```diff
@@ -0,0 +1,266 @@
+                    menuInflater.inflate(R.menu.autofill_simple_list_mode_menu, menu)
+                    searchMenuItem = menu.findItem(R.id.searchLogins)
+
+                    initializeSearchBar()
+                }
+
+                override fun onMenuItemSelected(menuItem: MenuItem): Boolean = false
+
+                override fun onPrepareMenu(menu: Menu) {
+                    val loginsSaved = !viewModel.viewState.value.logins.isNullOrEmpty()
+                    searchMenuItem?.isVisible = loginsSaved
+                }
+            },
+            viewLifecycleOwner,
+            Lifecycle.State.RESUMED,
+        )
+    }
+
+    private fun initializeSearchBar() {
+        searchMenuItem?.setOnMenuItemClickListener {
+            showSearchBar()
+            return@setOnMenuItemClickListener true
+        }
+
+        parentBinding()?.let { parentBinding ->
+            parentBinding.searchBar.onAction {
+                when (it) {
+                    is SearchBar.Action.PerformUpAction -> hideSearchBar()
+                    is SearchBar.Action.PerformSearch -> viewModel.onSearchQueryChanged(it.searchText)
+                }
+            }
+        }
+    }
+
+    private fun showSearchBar() = parentActivity()?.showSearchBar()
+    private fun hideSearchBar() = parentActivity()?.hideSearchBar()
+
+    private fun getRequestedUrl() = arguments?.getString(ARG_CURRENT_URL, null)
+    private fun getRequestedPackage() = arguments?.getString(ARG_CURRENT_PACKAGE, null)
+
+    private fun parentBinding() = parentActivity()?.binding
+    private fun parentActivity() = (activity as? AutofillProviderChooseActivity)
+
+    private fun observeViewModel() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.viewState.collect { state ->
+                    state.logins?.let {
+                        credentialsListUpdated(
+                            credentials = it,
+                            credentialSearchQuery = state.credentialSearchQuery,
+                        )
+                        parentActivity()?.invalidateOptionsMenu()
+                    }
+                }
+            }
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.viewState.collect {
+                    // we can just invalidate the menu as [onPrepareMenu] will handle the new visibility for importing passwords menu item
+                    parentActivity()?.invalidateOptionsMenu()
+                }
+            }
+        }
+
+        viewModel.onViewCreated()
+    }
+
+    private suspend fun credentialsListUpdated(
+        credentials: List<LoginCredentials>,
+        credentialSearchQuery: String,
+    ) {
+        if (credentials.isEmpty() && credentialSearchQuery.isEmpty()) {
+            showEmptyCredentialsPlaceholders()
```

Comment:
When there are no saved passwords, this UX is weird. Can’t see explicitly in Asana or Figma what the expectations are for empty state, but I think if there’s no saved passwords, we should prevent the user getting to this screen in the first place.

There’s no value in bringing them here just to tell them we’ve no saved passwords, and the buttons like “Import Passwords From Google” are just undesirable to see in this context.

![Screenshot_20250211_103414](https://github.com/user-attachments/assets/13f1e532-684c-4f26-9c50-7b73b634b4ec)

Reply:
Thanks for tidying up the empty state.

One last nit here: we can skip the auth prompt when passwords are empty like we do in the main password management screen. I already think it’s a questionable UX to let users visit this screen when we have no passwords to offer, but it’s a double insult to make them authenticate first to then tell them there’s nothing here for them 😄 

Feel free to add in later, though. 
![Screen Recording 2025-02-12 at 15 22 46](https://github.com/user-attachments/assets/9b160c95-e257-4c31-bedc-a45c7cd8f7f3)